lib/channels/src/index.ts

```diff
@@ -1,27 +1,60 @@
-/* eslint no-underscore-dangle: 0 */
+export interface ChannelTransport {
+  send: any; // todo Check actual type
+  setHandler<TEventArgs = any>(handler: (event: ChannelEvent<TEventArgs>) => void): void;
+}
+
+export interface ChannelEvent<TEventArgs = any> {
+  type: string;
+  from: string;
+  args: TEventArgs[];
+}
+
+export interface Listener<TEventArgs = any> {
+  (...args: TEventArgs[]): void;
+  ignorePeer?: boolean;
+}
+
+interface ListenersKeyValue {
+  [key: string]: Listener[];
+}
 
-export default class Channel {
-  constructor({ transport, async } = {}) {
-    this._sender = this._randomId();
+interface ChannelArgs {
+  transport?: ChannelTransport;
+  async?: boolean;
+}
+
+const generateRandomId = () => {
+  // generates a random 13 character string
+  return Math.random()
+    .toString(16)
+    .slice(2);
+};
+
+export class Channel {
+  private _sender = generateRandomId();
+  private _listeners: ListenersKeyValue = {};
+  private readonly _async: boolean = false;
```

Comment:
The readonly field that has a value can be changed in a ctro 😱 ??

Reply:
`readonly` can only be set directly or in the constructor, if you meant that?

Reply:
I just see this `_async` prop is also defined in ctor.